packages/dd-trace/src/exporters/agent/index.js

```diff
@@ -3,6 +3,7 @@
 const { URL, format } = require('url')
 const log = require('../../log')
 const Writer = require('./writer')
+const { fetchAgentInfo } = require('../common/util')
 
 class AgentExporter {
```

Comment:
There's sort of three states here. The first is that we don't know if the agent supports top level span events. The second is that we know it doesn't, and the third is that we know it does. So maybe we set a default value of null which is falsey to help with debugging.
```suggestion
class AgentExporter {
  _agentSupportsTopLevelSpanEvents = null
```
We need to decide if the _ prefix makes sense before applying this change.

Reply:
Also, any idea if there is a risk of shenanigans if messages are sent before the time it takes to get agent info?

Reply:
I don't think there's a problem if trace data is sent before the agent info is returned. In this case, the property would be considered falsey, so we would not do the top-level encoding.

Reply:
this is no longer relevant given that the setting is now an env configuration